contracts/utils/DAOTracker.sol

```diff
@@ -0,0 +1,92 @@
+pragma solidity ^0.5.4;
+
+import "@daostack/infra/contracts/Reputation.sol";
+import "../controller/DAOToken.sol";
+import "../controller/Avatar.sol";
+import "../controller/ControllerInterface.sol";
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+
+/**
+ * @title An on-chain "source of truth" for what DAOs
+ *        should be index into DAOstack's subgraph.
+ */
+contract DAOTracker is Ownable {
+
+    struct TrackingInfo {
+        DAOToken            nativeToken;
+        Reputation          nativeReputation;
+        ControllerInterface controller;
+
+        // `blacklist` this from the subgraph's cache
+        bool blacklist;
+    }
+
+    // Mapping from Avatar address to a flag letting the subgraph
+    // know if it should track this DAO or not.
+    mapping(address=>TrackingInfo) public tracking;
```

Comment:
does this storage is usage is a must ? 
the state can be track via the event

Reply:
Yes you're right. The situation where we'd want something on-chain is if we wanted to check the live state during a rebuild of the database. For example:
1. BadDAO adds itself
2. DAOstack blacklists BadDAO
3. The live subgraph receives the `BlacklistDAO` event and does its best to clear out all traces of BadDAO data.
4. BadDAO exploited a bug in the subgraph, so we need to rebuild the cache. When we rebuild and process BadDAO's `TrackDAO` event, we check the live state of the chain to see if it has been blacklisted, and if so we skip it.
```typescript
function handleTrackDAO(event: TrackDAO) {
  ...
  const info = daoTrackerSC.tracking(event._avatar).call({ block: "latest" })

  // Don't let BadDAO do what it did last time...
  if (info.blacklist) {
    return;
  }
}
```

Reply:
I've minimized the storage that's in use to just a `mapping(address=>boolean)` to support the above use case.